src/libutil/rust-ffi.hh

```diff
@@ -30,14 +30,29 @@ protected:
 
     // Must not be called directly.
     Value()
-    { }
+    {
+        // Precaution, in case this is used improperly
+        evacuate();
+    }
 
     Value(Value && other)
         : raw(other.raw)
     {
         other.evacuate();
     }
 
+    // Not all Rust types are Clone / Copy, but our base "value" class needs to
+    // have a copy constructor so the ones that are can be.
```

Comment:
"so the ones that are can be" sounds super duper omega wacky. Maybe something like:

```
    // Our base Value class needs to have a copy constructor so that Rust
    // types that implement Copy/Clone can be copied/cloned.
```

Reply:
```suggestion
    // Not all Rust types are Clone / Copy, but our base Value class needs to
    // have a copy constructor so that ones which do implement Copy/Clone
    // can be copied/cloned.
```

Reply:
"ones" is a bit too vague for my tastes.

"copy constructor, so types that do implement Copy/Clone can be copied/cloned." maybe?

---

Slightly more nit-picky: I'd smush the `Clone / Copy` to `Copy/Clone` (so it matches the second instance on the next line).